### PR TITLE
Update FTP, Kafka bbe contents

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/3804

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by praneesha — APPROVED** [review]

File: examples/ftp-client-read/ftp_client_read.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # FTP client - Read file
 
-The FTP client is used to perform CRUD operation on remote files/directories using the FTP protocol. This sample includes getting file content with default configurations using the default port number.
+The FTP client can be used to read files on remote servers using the FTP protocol. An `ftp:Client` can be created with default configurations and the `get` API can be used to read the files. This returns the file content as a `byte[]` stream which can then be written to another file/output to console etc.
```
Comment (shafreenAnfar) [inline examples/ftp-client-read/ftp_client_read.md:4]: **What does it do?**
The FTP client connects to a given FTP server, and then sends and receives files as byte streams.

**How to use it?** 
A FTP client is created by giving the host-name and required credentials. Once connected, `get` method is used to read files as byte streams from the FTP server.

**When to use it?**
Use this transfer files from a remote file system to a local file system.

Comment (dilanSachi) [reply examples/ftp-client-read/ftp_client_read.md:4]: Updated in https://github.com/ballerina-platform/ballerina-distribution/pull/3749/commits/ed9908a4875590a75415d06bda29db0c03360fe9

File: examples/kafka-client-consumer-ssl/kafka_client_consumer_ssl.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Kafka client - Consumer SSL/TLS
 
-This shows how the SSL encryption is done in the `kafka:Consumer`.
+The Kafka consumer connects to a Kafka server via SSL, and then receives payloads from the server. SSL can be enabled by providing the `kafka:SecureSocket` with the relevant values and `kafka:SecurityProtocol` as `kafka:PROTOCOL_SSL` in the `kafka:ConsumerConfiguration`. Use this to connect to a Kafka server secured with SSL.
```
Comment (shafreenAnfar) [inline examples/kafka-client-consumer-ssl/kafka_client_consumer_ssl.md:3]: SSL can be enabled by configuring the `secureSocket` which requires a certificate and the protocol name.

File: examples/kafka-client-consumer-sasl/kafka_client_consumer_sasl.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Kafka client - Consumer SASL authentication
 
-The `kafka:Consumer` connects to a Kafka server via SASL/PLAIN authentication, and then receives payloads from the server. SASL/PLAIN authentication is done by providing the `kafka:AuthenticationConfiguration` along with `kafka:SecurityProtocol` as `kafka:PROTOCOL_SASL_PLAINTEXT` in the `kafka:ConsumerConfiguration`. Use this to connect to a Kafka server secured with SASL/PLAIN.
+The `kafka:Consumer` connects to a Kafka server via SASL/PLAIN authentication, and then receives payloads from the server. SASL/PLAIN authentication can be enabled by configuring the `auth` which requires the authentication mechanism, username and a password. Further, the mode of security must be configured by setting securityProtocol to `kafka:PROTOCOL_SASL_PLAINTEXT`. Use this to connect to a Kafka server secured with SASL/PLAIN.
```
Comment (praneesha) [inline examples/kafka-client-consumer-sasl/kafka_client_consumer_sasl.md:3]:
```suggestion
The `kafka:Consumer` connects to a Kafka server via SASL/PLAIN authentication and then, receives the payloads from the server. SASL/PLAIN authentication can be enabled by configuring the `auth`, which requires the authentication mechanism, username, and password. Further, the mode of security must be configured by setting the `securityProtocol` to `kafka:PROTOCOL_SASL_PLAINTEXT`. Use this to connect to a Kafka server secured with SASL/PLAIN.
```

File: examples/kafka-client-consumer-ssl/kafka_client_consumer_ssl.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Kafka client - Consumer SSL/TLS
 
-The `kafka:Consumer` connects to a Kafka server via SSL, and then receives payloads from the server. SSL can be enabled by providing the `kafka:SecureSocket` with the relevant values and `kafka:SecurityProtocol` as `kafka:PROTOCOL_SSL` in the `kafka:ConsumerConfiguration`. Use this to connect to a Kafka server secured with SSL.
+The `kafka:Consumer` connects to a Kafka server via SSL/TLS, and then receives payloads from the server. SSL/TLS can be enabled by configuring the `secureSocket` which requires a certificate, and the protocol name. Further, the mode of security must be configured by setting `securityProtocol` to `kafka:PROTOCOL_SSL`. Use this to connect to a Kafka server secured with SSL/TLS.
```
Comment (praneesha) [inline examples/kafka-client-consumer-ssl/kafka_client_consumer_ssl.md:3]:
```suggestion
The `kafka:Consumer` connects to a Kafka server via SSL/TLS and then, receives payloads from the server. SSL/TLS can be enabled by configuring the `secureSocket`, which requires a certificate and the protocol name. Further, the mode of security must be configured by setting the `securityProtocol` to `kafka:PROTOCOL_SSL`. Use this to connect to a Kafka server secured with SSL/TLS.
```

File: examples/kafka-client-producer-sasl/kafka_client_producer_sasl.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # Kafka client - Producer SASL authentication
 
-The `kafka:Producer` connects to a Kafka server via SASL/PLAIN authentication, and then sends messages to the server. SASL/PLAIN authentication is done by providing the `kafka:AuthenticationConfiguration` along with `kafka:SecurityProtocol` as `kafka:PROTOCOL_SASL_PLAINTEXT` in the `kafka:ProducerConfiguration`. Use this to connect to a Kafka server secured with SASL/PLAIN.
+The `kafka:Producer` connects to a Kafka server via SASL/PLAIN authentication, and then sends messages to the server. SASL/PLAIN authentication can be enabled by configuring the `auth` which requires the authentication mechanism, username and a password. Further, the mode of security must be configured by setting securityProtocol to `kafka:PROTOCOL_SASL_PLAINTEXT`. Use this to connect to a Kafka server secured with SASL/PLAIN.
```
Comment (praneesha) [inline examples/kafka-client-producer-sasl/kafka_client_producer_sasl.md:3]:
```suggestion
The `kafka:Producer` connects to a Kafka server via SASL/PLAIN authentication and then, sends messages to the server. SASL/PLAIN authentication can be enabled by configuring the `auth`, which requires the authentication mechanism, username, and password. Further, the mode of security must be configured by setting the `securityProtocol` to `kafka:PROTOCOL_SASL_PLAINTEXT`. Use this to connect to a Kafka server secured with SASL/PLAIN.
```

File: examples/kafka-client-producer-ssl/kafka_client_producer_ssl.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # Kafka client - Producer SSL/TLS
 
-The `kafka:Producer` connects to a Kafka server via SSL, and then sends messages to the server. SSL can be enabled by providing the `kafka:SecureSocket` with the relevant values and `kafka:SecurityProtocol` as `kafka:PROTOCOL_SSL` in the `kafka:ProducerConfiguration`. Use this to connect to a Kafka server secured with SSL.
+The `kafka:Producer` connects to a Kafka server via SSL, and then sends messages to the server. SSL/TLS can be enabled by configuring the `secureSocket` which requires a certificate, and the protocol name. Further, the mode of security must be configured by setting `securityProtocol` to `kafka:PROTOCOL_SSL`. Use this to connect to a Kafka server secured with SSL.
```
Comment (praneesha) [inline examples/kafka-client-producer-ssl/kafka_client_producer_ssl.md:3]:
```suggestion
The `kafka:Producer` connects to a Kafka server via SSL and then, sends messages to the server. SSL/TLS can be enabled by configuring the `secureSocket`, which requires a certificate and the protocol name. Further, the mode of security must be configured by setting the `securityProtocol` to `kafka:PROTOCOL_SSL`. Use this to connect to a Kafka server secured with SSL.
```

File: examples/kafka-service-sasl/kafka_service_sasl.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # Kafka service - SASL authentication
 
-The `kafka:Service` receives messages from the Kafka server using the `kafka:Listener` via SASL/PLAIN authentication. SASL/PLAIN authentication is done by providing the `kafka:AuthenticationConfiguration` along with `kafka:SecurityProtocol` as `kafka:PROTOCOL_SASL_PLAINTEXT` in the `kafka:ConsumerConfiguration`. Use this to connect to a Kafka server secured with SASL/PLAIN.
+The `kafka:Service` receives messages from the Kafka server using the `kafka:Listener` via SASL/PLAIN authentication. SASL/PLAIN authentication can be enabled by configuring the `auth` which requires the authentication mechanism, username and a password. Further, the mode of security must be configured by setting securityProtocol to `kafka:PROTOCOL_SASL_PLAINTEXT`. Use this to connect to a Kafka server secured with SASL/PLAIN.
```
Comment (praneesha) [inline examples/kafka-service-sasl/kafka_service_sasl.md:3]:
```suggestion
The `kafka:Service` receives messages from the Kafka server using the `kafka:Listener` via SASL/PLAIN authentication. SASL/PLAIN authentication can be enabled by configuring the `auth`, which requires the authentication mechanism, username, and password. Further, the mode of security must be configured by setting the `securityProtocol` to `kafka:PROTOCOL_SASL_PLAINTEXT`. Use this to connect to a Kafka server secured with SASL/PLAIN.
```

File: examples/kafka-service-ssl/kafka_service_ssl.md [hunk 8]
```diff
@@ -1,6 +1,6 @@
 # Kafka service - SSL/TLS
 
-The `kafka:Service` receives messages from the Kafka server using the `kafka:Listener` via SSL. SSL can be enabled by providing the `kafka:SecureSocket` with the relevant values and `kafka:SecurityProtocol` as `kafka:PROTOCOL_SSL` in the `kafka:ConsumerConfiguration`. Use this to connect to a Kafka server secured with SSL.
+The `kafka:Service` receives messages from the Kafka server using the `kafka:Listener` via SSL. SSL/TLS can be enabled by configuring the `secureSocket` which requires a certificate, and the protocol name. Further, the mode of security must be configured by setting `securityProtocol` to `kafka:PROTOCOL_SSL`. Use this to connect to a Kafka server secured with SSL.
```
Comment (praneesha) [inline examples/kafka-service-ssl/kafka_service_ssl.md:3]:
```suggestion
The `kafka:Service` receives messages from the Kafka server using the `kafka:Listener` via SSL. SSL/TLS can be enabled by configuring the `secureSocket`, which requires a certificate and the protocol name. Further, the mode of security must be configured by setting the `securityProtocol` to `kafka:PROTOCOL_SSL`. Use this to connect to a Kafka server secured with SSL.
```

File: examples/ftp-service-read-write/ftp_service_read_write.md [hunk 9]
```diff
@@ -1,13 +1,13 @@
 # FTP service - Read/Write file
 
-The FTP service is used to receive file/directory changes that occur in a remote location using the FTP protocol. This sample includes receiving file/directory related change events from a listener and using the `append` api of the `ftp:Caller` to interact with the FTP server.
+The `ftp:Service` connects to a given FTP server via the `ftp:Listener` and then sends and receives files as byte streams via the `ftp:Caller`. The `ftp:Caller` is available as a parameter in the `onFileChange` method which is created with the configurations provided to the `ftp:Listener`. Once the `onFileChange` is invoked, the `append` method of FTP caller can be used to append a file to the files existing in the FTP server. Use this to listen to file changes occurring in a remote file system and do CRUD operations with the remote server.
```
Comment (shafreenAnfar) [inline examples/ftp-service-read-write/ftp_service_read_write.md:4]: The `ftp:Service` connects to a given FTP server via the `ftp:Listener`. Once connected, service starts receiving events every time a file deleted or added to the server. To take actions for these events `ftp:Caller` is used. The `ftp:Caller` can be specified as a parameter of `onFileChange` remote method. The `ftp:Caller` allows interacting with the server via `append`, `foo`, `bar`, etc remote methods. Use this to listen to file changes occurring in a remote file system and take action for those changes.

Comment (dilanSachi) [reply examples/ftp-service-read-write/ftp_service_read_write.md:4]: Fixed in https://github.com/ballerina-platform/ballerina-distribution/pull/3749/commits/6f93006b56680b05378562e2b52a31c73ff37fd6

File: examples/kafka-client-consumer-record-data-binding/kafka_client_consumer_record_data_binding.md [hunk 10]
```diff
@@ -1,8 +1,6 @@
...
 
-This shows how to use a `kafka:Consumer` as a simple record consumer. The records from a subscribed topic can be retrieved using the `poll()` function. This consumer uses the builtin byte array deserializer for both the key and the value, which is the default deserializer in the `kafka:Consumer`.
-
-The received records are converted to the user defined type using data-binding.
+The `kafka:Consumer` connects to a given Kafka server, and then receives messages as a `byte[]` or as a user defined type via data-binding. A `kafka:Consumer` is created by giving the server endpoint, group id and the topic. A user-defined type can be created by creating a subtype of `kafka:AnydataConsumerRecord` and overriding the type of `value` field. Once connected, `poll` method is used to receive messages with the intended type from the Kafka server. The received values will be bound to the target type internally. The consumer uses the built-in byte array deserializer for both the key and the value, which is the default deserializer in the `kafka:Consumer`. Use this to receive messages from a Kafka server with the metadata of the messages like `kafka:PartitionOffset` and `timestamp`.
```
Comment (shafreenAnfar) [inline examples/kafka-client-consumer-record-data-binding/kafka_client_consumer_record_data_binding.md:4]: **What does it do?**
The consumer record data-binding allows you to directly bind Kafka messages to subtypes `kafka:AnydataConsumerRecord` .  It does this by using the built-in bytes deserializer for both the key and value. 

**How to use it?**
To use this directly assign the `poll` method’s return value to the declared variable, which is a subtype of `kafka:AnydataConsumerRecord` . A subtype of `kafka:AnydataConsumerRecord` can be created by specifying a user defined type for the `value` field. 

**When to use it?**
Use this to receive messages from a Kafka server with the metadata of the messages like `kafka:PartitionOffset` and `timestamp`. It is important to note that this only works when `kafka:Producer` also uses the built-in bytes  serializer for Ballerina.

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**